ivc/src/ivc/mod.rs

```diff
@@ -7,32 +7,51 @@ mod tests {
 
     use crate::{
         ivc::{
-            columns::{IVCColumn, IVC_POSEIDON_NB_FULL_ROUND, IVC_POSEIDON_STATE_SIZE},
-            interpreter::ivc_circuit,
+            columns::{IVCColumn, IVC_POSEIDON_NB_FULL_ROUND, IVC_POSEIDON_STATE_SIZE, N_BLOCKS},
+            interpreter::{build_selectors, constrain_ivc, ivc_circuit},
             lookups::IVCLookupTable,
         },
         poseidon::{interpreter::PoseidonParams, params::static_params},
     };
     use ark_ff::{UniformRand, Zero};
-    use kimchi_msm::{circuit_design::WitnessBuilderEnv, columns::ColumnIndexer, Ff1, Fp};
+    use kimchi::circuits::domains::EvaluationDomains;
+    use kimchi_msm::{
+        circuit_design::{
+            composition::{IdMPrism, MPrism},
+            ConstraintBuilderEnv, SubEnvLookup, WitnessBuilderEnv,
+        },
+        columns::ColumnIndexer,
+        logup::LookupTableID,
+        precomputed_srs::get_bn254_srs,
+        proof::ProofInputs,
+        prover::prove,
+        verifier::verify,
+        witness::Witness,
+        BaseSponge, Ff1, Fp, OpeningProof, ScalarSponge, BN254,
+    };
+    use poly_commitment::pairing_proof::PairingSRS;
     use rand::{CryptoRng, RngCore};
 
     // Test number
-    pub const TEST_N_COL_TOTAL: usize = 50;
+    pub const TEST_N_COL_TOTAL: usize = IVCColumn::N_COL + 50;
+    // Absolutely no idea.
+    pub const TEST_N_CHALS: usize = 200;
     pub const TEST_DOMAIN_SIZE: usize = 1 << 15;
 
     #[derive(Clone)]
     pub struct PoseidonBN254Parameters;
 
-    type IVCWitnessBuilderEnv = WitnessBuilderEnv<
+    type IVCWitnessBuilderEnvRaw<LT> = WitnessBuilderEnv<
         Fp,
         IVCColumn,
         { <IVCColumn as ColumnIndexer>::N_COL },
         { <IVCColumn as ColumnIndexer>::N_COL },
         0,
         0,
-        IVCLookupTable<Ff1>,
+        LT,
     >;
+    //type IVCWitnessBuilderEnv = IVCWitnessBuilderEnvRaw<IVCLookupTable<Ff1>>;
```

Comment:
Nit: should we remove this? Or it is used later in a following PR?

Reply:
Will remove yes :)